proto/messages.proto

```diff
@@ -129,23 +139,40 @@ message Mention {
   required int32 start = 1; // offset from beginning of the message counting in utf16 characters
   required int32 length = 2;
   oneof mention_type {
+    // deprecated. Should be set such that old clients always fail when looking
+    // up the user. Ideally, this should not be a problem, as a non-federation
+    // aware user should never be part of a federated conversation.
     string user_id = 3;
   }
+  // only optional to maintain backwards compatibility.
+  optional QualifiedUser qualified_user = 4;
```

Comment:
The old field had an `..._id`. I would consider adding it to the new field too, i.e. `qualified_user_id`. This goes for all other IDs in the PR.

Reply:
No problem. I was trying to mimic the field names used by the new endpoints, but it might be better to be consistent within this document rather than be consistent across protocols.